consensus/helpers.go

```diff
@@ -1,58 +1,58 @@
 package consensus
 
+// TODO: Split this file into multiple helpers (e.g. signatures.go, hotstuff_helpers.go, etc...)
 import (
 	"encoding/base64"
-	"github.com/pokt-network/pocket/shared/debug"
 	"log"
 
+	"github.com/pokt-network/pocket/shared/codec"
+	"github.com/pokt-network/pocket/shared/debug"
+
 	"google.golang.org/protobuf/proto"
 
 	typesCons "github.com/pokt-network/pocket/consensus/types"
 	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
-	"google.golang.org/protobuf/types/known/anypb"
 )
 
 // These constants and variables are wrappers around the autogenerated protobuf types and were
 // added to simply make the code in the `consensus` module more readable.
-
 const (
 	NewRound  = typesCons.HotstuffStep_HOTSTUFF_STEP_NEWROUND
 	Prepare   = typesCons.HotstuffStep_HOTSTUFF_STEP_PREPARE
 	PreCommit = typesCons.HotstuffStep_HOTSTUFF_STEP_PRECOMMIT
 	Commit    = typesCons.HotstuffStep_HOTSTUFF_STEP_COMMIT
 	Decide    = typesCons.HotstuffStep_HOTSTUFF_STEP_DECIDE
 
+	Propose = typesCons.HotstuffMessageType_HOTSTUFF_MESSAGE_PROPOSE
+	Vote    = typesCons.HotstuffMessageType_HOTSTUFF_MESSAGE_VOTE
+
 	ByzantineThreshold = float64(2) / float64(3)
-	HotstuffMessage    = "consensus.HotstuffMessage"
-	UtilityMessage     = "consensus.UtilityMessage"
-	Propose            = typesCons.HotstuffMessageType_HOTSTUFF_MESAGE_PROPOSE
-	Vote               = typesCons.HotstuffMessageType_HOTSTUFF_MESSAGE_VOTE
+
+	HotstuffMessage = "consensus.HotstuffMessage"
+	UtilityMessage  = "consensus.UtilityMessage"
 )
 
 var (
 	HotstuffSteps = [...]typesCons.HotstuffStep{NewRound, Prepare, PreCommit, Commit, Decide}
-
-	maxTxBytes        = 90000             // TODO(olshansky): Move this to config.json.
-	lastByzValidators = make([][]byte, 0) // TODO(olshansky): Retrieve this from persistence
 )
 
 // ** Hotstuff Helpers ** //
 
+// IMPROVE: Avoid having the `ConsensusModule` be a receiver of this; making it more functional.
+// TODO: Add unit tests for quorumCert creation & validation.
 func (m *ConsensusModule) getQuorumCertificate(height uint64, step typesCons.HotstuffStep, round uint64) (*typesCons.QuorumCertificate, error) {
 	var pss []*typesCons.PartialSignature
 	for _, msg := range m.MessagePool[step] {
-		// TODO(olshansky): Add tests for this
 		if msg.GetPartialSignature() == nil {
 			m.nodeLog(typesCons.WarnMissingPartialSig(msg))
 			continue
 		}
-		// TODO(olshansky): Add tests for this
-		if msg.Height != height || msg.Step != step || msg.Round != round {
+		if msg.GetHeight() != height || msg.GetStep() != step || msg.GetRound() != round {
```

Comment:
noop: again, I would love a function here that returns a bool.

I am about to go on a tangent... I'd love your thoughts on this.

The more I see these inside the consensus module and the more I come to understand/believe that we should probably have some sort of "state machine" that given the current "state" exposes some validation logic that could be made quite generic.

This state machine or `consensusMessageValidatorThinghy`™️ would be injected in the consensus module and the validation itself would be an interface method for which the inputs would be the hotstuff message and the current "state" tuple made of height, step, round and whatever else is needed to perform these validations.

I guess this would be an issue on itself.